### PR TITLE
Optionally allow a different branch for each repo

### PR DESCRIPTION
**Pull request by jvlcek**
https://github.com/ManageIQ/manageiq-appliance-build/issues/95

This PR will make building development image to test proposed changes simpler.
This will be done by adding optional command line arguments where specific branch names
can be specified for each of the repos.

No default behavior is changing.

### CODE REVIEW

File: scripts/cli.rb [hunk 1]
```diff
@@ -29,13 +29,16 @@ def parse(args = ARGV)
         banner "Usage: build.rb [options]"
 
         opt :type,          type_desc,      :type => :string,  :short => "t", :default => DEFAULT_TYPE
-        opt :reference,     git_ref_desc,   :type => :string,  :short => "r", :default => DEFAULT_REF
         opt :local,         local_desc,     :type => :boolean, :short => "l", :default => false
-        opt :fileshare,     share_desc,     :type => :boolean, :short => "s", :default => true
+        opt :fileshare,     share_desc,     :type => :boolean, :short => "f", :default => true
```
Comment (chessbyte) [inline scripts/cli.rb]: should we alphabetize this list of options for clarity?

**Comment by jvlcek** [comment]
@abellotti Please review.
I have used these changes to successfully build a test image with the command of the form:


`ruby /build/scripts/vmbuild.rb --type nightly --only openstack -b branch_name -B https://github.com/jvlcek/manageiq-appliance-build.git`

File: scripts/cli.rb (hunk 1 above)
Comment (jvlcek) [inline scripts/cli.rb]: @chessbyte :+1: Willdo

**Comment by jvlcek** [comment]
I'm working on fixing the travis failure now.

**Comment by abellotti** [comment]
Hi Joe, looks good.

l think we still need to support the -r option for compatibility, we have scripts that make use of that option,  in the code if -r is specified, set to -m, -a and -b (if not specified).  That will also make it a nice shortcut if we really want all three to be the same.

Thanks for the quick turnaround on this !!

**Comment by jvlcek** [comment]
@abellotti Thank you. Yes I had reviewed this with @simaishi  and she also suggested maintaining the -r flag. Which I have now done.

File: scripts/cli.rb [hunk 2]
```diff
@@ -46,11 +49,19 @@ def parse(args = ARGV)
...
+      # --reference overrides all other reference arguments
+      unless options[:reference].to_s.empty?
+        options[:manageiq_ref]  = options[:reference]
+        options[:appliance_ref] = options[:reference]
+        options[:build_ref]     = options[:reference]
+        options[:ssui_ref]      = options[:reference]
+      end
+
+      Trollop.die(:manageiq_ref, git_ref_desc) if options[:manageiq_ref].to_s.empty?
+      options[:manageiq_ref] = options[:manageiq_ref].to_s.strip
```
Comment (kbrock) [inline scripts/cli.rb]: I think you want to move this to 53.5 - before using reference to set all the other references

Comment (jvlcek) [inline scripts/cli.rb]: :+1: Thank you @kbrock

**Comment by kbrock** [comment]
LGTM :+1:

**Pull request merged**